[video_player_platform_interface] synchronize isPlaying state

maRci002 opened the pull request:
This PR is part of #3261

This is why this part is created: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changing-federated-plugins

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

bparrishMines reviewed (commented):
@stuartmorgan It looks like this is causing analyze errors because of the enum change. Based on https://github.com/flutter/flutter/issues/89866#issuecomment-1380549377, should this temporarily add 

```
// ignore: missing_enum_constant_in_switch
```

to the implementation packages.

stuartmorgan commented on packages/video_player/video_player_platform_interface/pubspec.yaml:
```diff
@@ -4,7 +4,7 @@ repository: https://github.com/flutter/packages/tree/main/packages/video_player/
 # NOTE: We strongly prefer non-breaking changes, even at the expense of a
 # less-clean API. See https://flutter.dev/go/platform-interface-breaking-changes
-version: 6.0.2
+version: 6.0.3
```
This is new API, so should be 6.1.0.

bparrishMines commented:
@maRci002 As mentioned above, can you add `// ignore: missing_enum_constant_in_switch` to the implementations where you are seeing the analyzer test fail?

maRci002 commented:
@bparrishMines I've added `// ignore: missing_enum_constant_in_switch` to avoid analyzer fails.

Now I've some errors in the CHANGELOG in the following packages: `video_player` / `video_player_android` / `video_player_avfoundation`. I think I should not make changes in the CHANGELOG since I just added lint ignore and #3261 will remove them.

stuartmorgan commented:
Overriding version and changelog checks: the changes are `// ignore`s, which are exempt.

stuartmorgan commented:
This will need to wait for https://github.com/flutter/packages/pull/3417 to land and then be re-merged, to avoid breaking post-submit.

maRci002 commented:
@stuartmorgan I've re-merged the newest master.

`video_player` / `video_player_android` / `video_player_avfoundation` packages throws:

> Dart changes are not allowed to other packages in video_player in the same PR as changes to public Dart code in video_player_platform_interface, as this can cause accidental breaking changes to be missed by automated checks. Please split the changes to these two packages into separate PRs.
> 
> If you believe that this is a false positive, please file a bug.

stuartmorgan commented:
I expected that, but misremembered and thought we had a way to override it in presubmit (which is why I suggested waiting rather than splitting it initially, sorry about that).

I filed https://github.com/flutter/flutter/issues/122390 for the false positive, but for now you can just split the ignores into a new PR. We can get that landed quickly, unblocking this one.

maRci002 commented:
> I filed [flutter/flutter#122390](https://github.com/flutter/flutter/issues/122390) for the false positive, but for now you can just split the ignores into a new PR. We can get that landed quickly, unblocking this one.

I created the ignore PR: #3435, when it is merged, then I will update this branch and remove the ignores in this PR.

stuartmorgan commented:
Removing the overrides since those changes will be moving to the other PR.

maRci002 commented:
I have updated this PR to not touch ignores meanwhile updated #3261 to remove lint checks.

BraveEvidence commented:
This will help https://www.youtube.com/watch?v=IMQdSTlTXjA

stuartmorgan reviewed (approved):
LGTM

bparrishMines reviewed (approved):
LGTM
